### PR TITLE
Oval crop corners added

### PR DESCRIPTION
**Pull request by bhagyae5308**
The changes are related to https://github.com/CanHub/Android-Image-Cropper/issues/305

1. Added provision to add Oval crop corners when the crop style is Rectangle
2. Setters exposed to had crop corner radius and fill color

### CODE REVIEW

File: cropper/src/main/java/com/canhub/cropper/CropImageContractOptions.kt [hunk 1]
```diff
@@ -49,6 +49,30 @@ data class CropImageContractOptions @JvmOverloads constructor(
...
+     */
+    fun setCropCornerShape(cornerShape: CropImageView.CropCornerShape): CropImageContractOptions {
+         cropImageOptions.cornerShape = cornerShape
+        return this
+    }
+
+    /**
+     * To set the fill color of the Oval crop corner
+     */
+    fun setCircleCornerFillColor(circleFillColor:Int):CropImageContractOptions{
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/CropImageContractOptions.kt:64]: Do we expect the value on RGB or Hex or Android Int? Would be nice to have some tip on the variable name, like `circleFillColorHexValue` and to have it clear on the documentation

File: cropper/src/main/java/com/canhub/cropper/CropImageContractOptions.kt [hunk 2]
```diff
@@ -49,6 +49,30 @@ data class CropImageContractOptions @JvmOverloads constructor(
...
+        cropImageOptions.circleCornerFillColor = circleFillColor
+        return this
+    }
+
+    /**
+     * To set the Oval crop corner radius
+     */
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/CropImageContractOptions.kt:71]: Default value?

File: cropper/src/main/java/com/canhub/cropper/CropOverlayView.kt [hunk 3]
```diff
@@ -49,7 +50,18 @@ class CropOverlayView
                 borderPaint.isAntiAlias = true
                 borderPaint
             } else null
+        private fun getNewPaintWithFill(color: Int): Paint? {
+            val borderPaint = Paint()
+            borderPaint.color = color
+            borderPaint.style = Paint.Style.FILL
+            // borderPaint.isAntiAlias = true
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/CropOverlayView.kt:57]: ?

File: cropper/src/main/java/com/canhub/cropper/CropOverlayView.kt [hunk 4]
```diff
@@ -297,6 +324,10 @@ class CropOverlayView
     fun setSnapRadius(snapRadius: Float) {
         mSnapRadius = snapRadius
     }
+    //ToDO:Need to add documentation
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/CropOverlayView.kt:327]: 👀 you said hahahaha

**Review by Canato — COMMENTED** [review]
Didn't finish the review yet, will give it more attention tomorrow probably, but great work =D

Thanks for this, raise some minor stuffs

**Comment by Canato** [comment]
@bhagyae5308 the CI is not passing, could you take a look please?

**Comment by bhagyae5308** [comment]
@Canato Apologies for late update, I was occupied with some other tasks.

File: cropper/src/main/java/com/canhub/cropper/CropImageContractOptions.kt [hunk 5]
```diff
@@ -49,6 +49,32 @@ data class CropImageContractOptions @JvmOverloads constructor(
...
+     * Default: RECTANGLE
+     */
+    fun setCropCornerShape(cornerShape: CropImageView.CropCornerShape): CropImageContractOptions {
+        cropImageOptions.cornerShape = cornerShape
+        return this
+    }
+
+    /**
+     * To set the fill color of the Oval crop corner
+     * @param circleFillColorHexValue Hex value of the color (Default is MAGENTA)
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/CropImageContractOptions.kt:63]: Let's change the default to White please

File: cropper/src/main/java/com/canhub/cropper/CropOverlayView.kt [hunk 6]
```diff
@@ -531,6 +568,8 @@ class CropOverlayView
                 canvas
             )
         }
+        // To retain the changes in Paint object when the App goes background this is required
+        mBorderCornerPaint = getNewPaintOrNull(mOptions?.borderCornerThickness ?: 0.0f, mOptions?.borderCornerColor ?: Color.WHITE)
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/CropOverlayView.kt:572]: Cool

File: sample/src/main/res/layout/fragment_options.xml [hunk 7]
```diff
@@ -52,6 +52,9 @@
             <include
                 android:id="@+id/cropShape"
                 layout="@layout/chip_crop_shape" />
+            <include android:id="@+id/cornerShape"
+                layout="@layout/chip_corner_shape"
+                android:visibility="gone"/>
```
Comment (Canato) [inline sample/src/main/res/layout/fragment_options.xml:57]: Nice touch. Can see you are a detailed dev, thanks for it

**Review by Canato — CHANGES_REQUESTED** [review]
No worries about the time, all work here is volunteer you are already adding great value to the lib! and i'm thankful for this!

Now I had time to check with more attention ^^

1. We need to update the CHANGELOG

Almost ready to go 🎉

File: cropper/src/main/java/com/canhub/cropper/CropOverlayView.kt [hunk 8]
```diff
@@ -908,28 +908,28 @@ class CropOverlayView
     ) {
         // Top left
         canvas.drawCircle(
-            rect.left,
+            rect.left - cornerExtension,
             (rect.top - cornerExtension),
```
Comment (bhagyae5308) [inline cropper/src/main/java/com/canhub/cropper/CropOverlayView.kt:912]: Added this change to align properly the corners over the edges of the rectangle. Observed this while checking it for smaller radius.

**Review by Canato — APPROVED** [review]

**Pull request merged**